packages/nightingale-msa/src/workers/conservation-inline-worker.ts

```diff
@@ -40,16 +39,16 @@ export const calculateConservation = (
   if (isWorker) {
     self.postMessage({ progress: 1, conservation }, [conservation.buffer]);
   }
-
   return conservation;
 };
 
-const onmessage = function (e: MessageEvent) {
+const conservationInlineWorkerString = `
+self.addEventListener('message', (e) => {
   if (self.previous !== e.data) {
-    calculateConservation(e.data.sequences, e.data.sampleSize, true);
+    const f = ${calculateConservation.toString()};
```

Comment:
🤯
wow, I'm really amazed by this trick!

Reply:
I have to agree it was surprising when it worked!